### PR TITLE
feat: trace level logging

### PR DESCRIPTION
**Pull request by njlie**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -78,7 +83,7 @@ class Plugin extends BtpPlugin {
   sendTransfer () {}
 
   async _createOutgoingChannel () {
-    debug('creating outgoing channel')
+    this._log.trace('creating outgoing channel')
```
Comment (sharafian) [inline index.js]: Creating an outgoing channel is an important step so this should probably be `info`

File: index.js [hunk 2]
```diff
@@ -88,25 +93,25 @@ class Plugin extends BtpPlugin {
...
     // TODO: make a generic version of the code that submits these things
-    debug('waiting for transaction to be added to the ledger')
+    this._log.trace('waiting for transaction to be added to the ledger')
     return new Promise((resolve) => {
       const handleTransaction = (ev) => {
         if (ev.transaction.SourceTag !== txTag) return
         if (ev.transaction.Account !== this._address) return
 
-        debug('transaction complete')
+        this._log.trace('transaction complete')
```
Comment (sharafian) [inline index.js]: It's good to see when the transaction is complete, so this can be `info` too

File: index.js [hunk 3]
```diff
@@ -415,16 +420,16 @@ class Plugin extends BtpPlugin {
...
         )
       } catch (err) {
-        debug('verifying signature failed:', err.message)
+        this._log.debug('verifying signature failed:', err.message)
       }
 
       if (!isValid) {
         // TODO: if these get out of sync, all subsequent transfers of money will fail
-        debug('invalid claim signature for', dropAmount)
+        this._log.debug('invalid claim signature for', dropAmount)
```
Comment (sharafian) [inline index.js]: I'd make this an error because it means the connector might be up to something

File: index.js [hunk 4]
```diff
@@ -476,12 +481,12 @@ class Plugin extends BtpPlugin {
...
         })
         .catch((e) => {
-          debug('fund tx/notify failed:', e)
+          this._log.error('fund tx/notify failed:', e)
           this._funding = false
         })
     }
 
-    debug('setting new claim. amount=' + amount)
+    this._log.info('setting new claim. amount=' + amount)
```
Comment (sharafian) [inline index.js]: this will happen on almost every packet so we can probably put it on the debug level

Comment (njlie) [reply index.js]: This seems trivial enough to me that it could go into trace, especially if it may spam over more valuable debug output. Thoughts?

Comment (sharafian) [reply index.js]: sure, trace sounds ok

**Pull request merged**